docs-gen/content/docs/examples/pipeline_dependencies.md

```diff
@@ -0,0 +1,105 @@
+---
+title: "Pipeline Dependencies"
+weight: 2
+---
+
+Pipeline dependencies allow splitting up larger machine learning pipelines into sub-pipelines. This is particularly useful when:
+- The data of an earlier step changes at a lower frequency than the data for subsequent steps
+- Outputs of an earlier step could be shared between pipelines to avoid re-processing the same data
+
+In this example, we break up the penguin example pipeline into two pipelines:
+- The *Penguin Examples Pipeline* has a single pipeline step that imports the CSV example and outputs it as an artifact
+- The *Penguin Training Pipeline* references the previously produces example and trains the model 
```

Comment:
Maybe add that in this case it doesn't have any practical use, but it's for demonstration purposes.